source/code/troubleshooter/omsagent_tsg_zip/tsg_files/tsg_code/tsg_general.py

```diff
@@ -0,0 +1,191 @@
+
+# check to make sure the user is running as root
+def check_sudo():
+    if (os.geteuid() != 0):
+        print("The troubleshooter is not currently being run as root. In order to "\
+              "have accurate results, we ask that you run this troubleshooter as root.\n"\
+              "The OMS Agent Troubleshooter needs to be run as root for the following reasons:")
+        print("  - getting workspace ID and other relevant information to debugging")
+        print("  - checking files in folders with strict permissions")
+        print("  - checking certifications exist / are correct")
+        # TODO: add more reasons as troubleshooter changes
+        print("NOTE: it will not add, modify, or delete any files without express permission.")
+        print("Please try running the troubleshooter again with 'sudo'. Thank you!")
+        return False
+    else:
+        return True
+
+
+
+# run through all troubleshooting scenarios
+def check_all():
+    all_success = 0
+    # 1: Install
+    checked_install = check_installation()
+    if (is_error(checked_install)):
+        return checked_install
+    else:
+        all_success = checked_install
+    
+    print("================================================================================")
+    # 2: Connection
+    checked_connection = check_connection()
+    if (is_error(checked_connection)):
+        return checked_connection
+    else:
+        all_success = checked_connection
+
+    print("================================================================================")
+    # 3: Heartbeat
+    checked_hb = check_heartbeat()
+    if (is_error(checked_hb)):
+        return checked_hb
+    else:
+        all_success = checked_hb
+
+    print("================================================================================")
+    checked_highcpumem = check_high_cpu_memory()
+    if (is_error(checked_highcpumem)):
+        return checked_highcpumem
+    else:
+        all_success = checked_highcpumem
+
+    print("================================================================================")
+    checked_syslog = check_syslog()
+    if (is_error(checked_syslog)):
+        return checked_syslog
+    else:
+        all_success = checked_syslog
+
+    print("================================================================================")
+    checked_cl = check_custom_logs()
+    if (is_error(checked_cl)):
+        return checked_cl
+    else:
+        all_success = checked_cl
+
+    return all_success
+
+
+
+
+def collect_logs():
+    # get SR number / company name
+    print("Please input the SR number to collect OMS logs and (if applicable) the company\n"\
+        "name for reference. (Leave field empty to skip)")
+    sr_num = get_input("SR Number", (lambda x : True), "")
+    com_name = get_input("Company Name", (lambda x : True), "")
+
+    # create command to run
+    logcollect_cmd = ['sudo', 'sh', logcollect_sh_path]
+    if (sr_num != ''):
```

Comment:
isn't SR number mandatory?

Reply:
I didn't realize it was mandatory, sorry! Updated the log collector to make it mandatory:
https://github.com/microsoft/OMS-Agent-for-Linux/blob/bff7c80f869d6f5a4046ed41b4bd0ed89d69219f/source/code/troubleshooter/modules/main.py#L116